### PR TITLE
Add routing to admin portal

### PR DESCRIPTION
**Pull request by tsatam**
### Which issue this PR addresses:

Fixes [ARO-3500](https://issues.redhat.com/browse/ARO-3500)

### What this PR does / why we need it:

Changes the URL of the Admin portal based on navigation state:
- Cluster details will be at https://<portal_url:port>/<resource_id>
- Routing for sub-panes within the cluster details is also present as a suffix on the route (e.g. `/overview`).
- pkg/portal/portal.go updated to route the above routes to the frontend's index.html

Details for an individual cluster can be routed to directly and linked to. 

### Test plan for issue:
- Existing E2E tests should continue to pass
- Added routing checks to E2E checks to validate expected route changes are made
- Added explicit E2E tests to ensure opening the page directly to a route works as expected

### Is there any documentation that needs to be updated for this PR?

N/A

### Additional Notes

- This PR adds the [React Router](https://reactrouter.com) library to the admin portal NPM dependencies in order to implement routing. 
- This PR changes the `<OverviewComponent>` to display cluster properties in a Fluent UI `<DetailsList>`, as well as changes the presentation logic to display all known properties whether they are present on the given cluster or not. 
  - This was done to fix the flaky end-to-end test on this component, blocking the merge of this and many other PRs.

### CODE REVIEW

**Comment by tsatam** [comment]
TODOs discussed during sync review:
- [ ] Add handling for authentication redirects to preserve the route. This is necessary in order for direct links to work correctly when not previously authenticated. 
- [ ] E2E tests should validate both the setting of the expected URL in different UI states, as well as the re-population of those UI states when freshly navigated directly to their corresponding URLs. 
- [ ] (nice-to-have) Populating the contents of the search bar into a `search` query parameter, allowing search results to be shared

**Review by rhamitarora — DISMISSED** [review]

File: portal/v2/src/ClusterDetailListComponents/Overview.tsx [hunk 1]
```diff
@@ -82,71 +28,37 @@ const clusterDetailHeadings : IClusterDetails = {
...
+        Object.entries(clusterDetailHeadings)
+            .map(([property, heading], index) => 
+                ({key: index, name: heading, value: props.item[property]})
+            ), 
+        [props.item])
+
+    return (
+        <Stack styles={contentStackStylesNormal}>
+            <Stack horizontal>
+                <ShimmeredDetailsList
```
Comment (lranjbar) [inline portal/v2/src/ClusterDetailListComponents/Overview.tsx:49]: +1 For bringing in premade React components from [fluentui](https://github.com/microsoft/fluentui).

**Review by lranjbar — APPROVED** [review]
+1 Routing is a hard thing to bring in if not done from the start but this looks like a good step!

**Review by bennerv — APPROVED** [review]
looks good to me

**Review by s-amann — APPROVED** [review]

**Review by rhamitarora — APPROVED** [review]

**Pull request merged**